subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -347,6 +347,10 @@ static void send_pending_adv(struct k_work *work)
 
 	if (atomic_test_bit(ext_adv->flags, ADV_FLAG_SUSPENDING)) {
 		LOG_DBG("Advertiser is suspending");
+
+		if (ext_adv->adv) {
+			bt_mesh_adv_unref(ext_adv->adv);
```

Comment:
I think this will not release all other adv buffers pending after the one that was scheduled. I've prepared a draft when was discussing this issue with @akredalen with the test that checks this behavior:
https://github.com/zephyrproject-rtos/zephyr/compare/main...PavelVPV:experiment_adv_disable_issue?expand=1#diff-96959e58e40e61bba9cc068e83e7157908d569a5282c573ff55d70d1e99755c9R584